http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -1194,6 +1209,94 @@ class GKEInterpreter[F[_]](
     } yield ()
   }
 
+  private[util] def installRStudioApp(
+    helmAuthContext: AuthContext,
+    appName: AppName,
+    release: Release,
+    cluster: KubernetesCluster,
+    nodepoolName: NodepoolName,
+    namespaceName: NamespaceName,
+    disk: PersistentDisk,
+    ksaName: ServiceAccountName,
+    gsa: WorkbenchEmail,
+    userEmail: WorkbenchEmail
+  )(implicit ev: Ask[F, AppContext]): F[Unit] = {
+    val chart = config.RStudioAppConfig.chart
+
+    for {
+      ctx <- ev.ask
+
+      _ <- logger.info(ctx.loggingCtx)(
+        s"Installing helm chart for RStudio app ${appName.value} in cluster ${cluster.getClusterId.toString}"
+      )
+
+      googleProject <- F.fromOption(
+        LeoLenses.cloudContextToGoogleProject.get(cluster.cloudContext),
+        new RuntimeException("trying to create an azure runtime in GKEInterpreter. This should never happen")
+      )
+
+      // Create the staging bucket to be used by Welder
+      stagingBucketName = generateUniqueBucketName("leostaging-" + appName.value)
+
+      _ <- bucketHelper
+        .createStagingBucket(userEmail, googleProject, stagingBucketName, gsa)
+        .compile
+        .drain
+
+      chartValues = buildRStudioAppChartOverrideValuesString(appName,
+                                                             cluster,
+                                                             nodepoolName,
+                                                             namespaceName,
+                                                             disk,
+                                                             ksaName,
+                                                             userEmail,
+                                                             stagingBucketName
+      )
+      _ <- logger.info(ctx.loggingCtx)(s"Chart override values are: $chartValues")
+
+      // Invoke helm
+      helmInstall = helmClient
+        .installChart(
+          release,
+          chart.name,
+          chart.version,
+          org.broadinstitute.dsp.Values(chartValues.mkString(",")),
+          false
+        )
+        .run(helmAuthContext)
+
+      // Currently we always retry.
+      // The main failure mode here is helm install, which does not have easily interpretable error codes
+      retryConfig = RetryPredicates.retryAllConfig
+      _ <- tracedRetryF(retryConfig)(
+        helmInstall,
+        s"helm install for RSTUDIO app ${appName.value} in project ${cluster.cloudContext.asString}"
+      ).compile.lastOrError
+
+      googleProject <- F.fromOption(
+        LeoLenses.cloudContextToGoogleProject.get(cluster.cloudContext),
+        new RuntimeException("trying to create an azure runtime in GKEInterpreter. This should never happen")
+      )
+      // Poll the app until it starts up
+      last <- streamFUntilDone(
+        config.RStudioAppConfig.services
+          .map(_.name)
+          .traverse(s => appDao.isProxyAvailable(googleProject, appName, s)),
```

Comment:
Yeah related to above comment -- I think here it's just using `HttpAppDAO` for polling, not `HttpRStudioAppDAO`.

Reply:
Good catch, thanks!

Reply:
To be clear I think this code is doing the right thing, just was commenting re usage of HttpAppDAO vs HttpRstudioAppDAO

Reply:
But maybe I should switch to using the HttpRstudioAppDAO get status instead?

Reply:
No strong preference, I think `HttpAppDAO` is probably fine. I think they are both doing the same thing, just hit the root URL and check for an ok http status.